### DISCUSSION TITLE
How often does the SOC update?

### DISCUSSION THREAD

**Discussion by iambenmitchell** [pypolestar/polestar_api#131] [General]
Hey there, I'm thinking of purchasing a 2nd hand polestar 2 soon and here in the UK I am using an energy tariff that changes every 30 minutes called octopus agile. 

The benefits is that the regular energy rate most people pay is around d £0.27/kWh. But the rate I pay can end up being negative, meaning I get PAID to use electrify, on the flip side, it can be as high as £1/kWh. 

I'm already using home assistant for controlling the radiators in each room of the house and so as a programmer myself, it would be easy for me to create some logic for charging the car at the cheapest rates possible using the API from the charger + polestar

For example:

if soc < 30% then charge to 30%
else
if energy_price < £0.10 then charge to 60% 
and if energy_price < £0.02 then charge to 100% 

However, to do this reliably I need an accurate SOC reading from the car. And I'd need it to be relatively up to date. 

Essentially what I am asking is, how often does the SOC figure from this API update? Once a minute?, 5 minutes? hour?

Thanks in advanced! And thank you for laying the foundations of integrating this into cool automations :)

**Comment by wouterhardeman** [comment]
The integration fetches data every 30 seconds I believe. In my experience it's safe to assume that the SoC updates every 5 minutes. 

Although in some instances I have seen a delay of about 30 minutes. I.e. after 30 minutes it starts updating as if it was 30 minutes ago. So you see a rising SoC every 30 seconds but it's 30 minutes late. This is very rare though.
